internal/auth0/prompt/data_source_screen_partials_test.go

```diff
@@ -0,0 +1,76 @@
+package prompt_test
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/auth0/terraform-provider-auth0/internal/acctest"
+)
+
+const testAccPromptScreenPartialsInvalid = `
+data "auth0_prompt_screen_partials" "prompt_screen_partials" {
+	  prompt_type = "login-xxxxx"
+}
+`
+
+const testAccPromptScreenPartialsWithoutScreens = testAccGivenACustomDomain + testGivenABrandingTemplate + `
+data "auth0_prompt_screen_partials" "prompt_screen_partials" {
+	  prompt_type = "login-passwordless"
+}
+`
+
+const testAccPromptScreenPartialsData = testAccGivenACustomDomain + testGivenABrandingTemplate + `
+resource "auth0_prompt_screen_partials" "prompt_screen_partials" {
+	  depends_on = [ auth0_branding.my_brand ]
+	  prompt_type = "login-passwordless"
+	  screen_partials {
+			screen_name = "login-passwordless-email-code"
+			insertion_points {
+				form_content_start = "<div>Form Content Start</div>"
+				form_content_end = "<div>Form Content End</div>"
+			}
+	  }
+}
+
+data "auth0_prompt_screen_partials" "prompt_screen_partials" {
+  	  depends_on = [ auth0_branding.my_brand ]
+	  prompt_type = auth0_prompt_screen_partials.prompt_screen_partials.prompt_type
+}
```

Comment:
Ideally, this should depend on `auth0_prompt_screen_partials.prompt_screen_partials` instead of the `auth0_branding.my_brand`